packages/graphiql/src/YogaGraphiQL.tsx

```diff
@@ -144,6 +148,7 @@ export function YogaGraphiQL(props: YogaGraphiQLProps): React.ReactElement {
     false,
   )
   const [showDocs, setShowDocs] = React.useState(false)
+  const [headers, setHeaders] = React.useState(props.headers || '{}')
```

Comment:
Wouldnt this conflict and overwrite the `shouldPersistHeaders` option? https://github.com/graphql/graphiql/blob/7f695b104f9b25ba8c6d36f7827c475b297b7482/packages/graphiql/src/components/GraphiQL.tsx#L193-L197

Reply:
Yeah the default case we need to extract from local storage

Reply:
This should be solved within GraphiQL not here.

Reply:
Ok removed it. Other than that, I think `additionalHeaders` makes sense.